tests/ltp/patches/setsid01.patch

```diff
@@ -0,0 +1,260 @@
+diff --git a/testcases/kernel/syscalls/setsid/setsid01.c b/testcases/kernel/syscalls/setsid/setsid01.c
+index ed8f0e4ad..67f086ad7 100644
+--- a/testcases/kernel/syscalls/setsid/setsid01.c
++++ b/testcases/kernel/syscalls/setsid/setsid01.c
+@@ -39,6 +39,7 @@
+ #include <sys/types.h>
+ #include <sys/stat.h>
+ #include "test.h"
++#include <pthread.h>
+ 
+ #define INVAL_FLAG	-1
+ #define USER2		301
+@@ -53,161 +54,113 @@ int TST_TOTAL = 1;
+ static char *argv0;
+ #endif
+ 
+-void do_child_1(void);
+-void do_child_2(void);
+ void setup(void);
+ void cleanup(void);
+ 
++static void* do_child_thread(void* arg);
++static void* do_master_thread(void* arg);
++
++
+ int main(int ac, char **av)
+ {
+-	int pid;
+-	int fail = 0;
+-	int ret, status;
+-	int exno = 0;
+-
+-	int lc;
++  int fail = 0;
++  int lc;
+ 
+-	tst_parse_opts(ac, av, NULL, NULL);
++  tst_parse_opts(ac, av, NULL, NULL);
+ #ifdef UCLINUX
+-	argv0 = av[0];
++  argv0 = av[0];
+ 
+-	maybe_run_child(&do_child_1, "n", 1);
+-	maybe_run_child(&do_child_2, "n", 2);
++  maybe_run_child(&do_child_1, "n", 1);
++  maybe_run_child(&do_child_2, "n", 2);
+ #endif
+ 
+-	/*
+-	 * perform global setup for the test
+-	 */
+-	setup();
+-
+-	for (lc = 0; TEST_LOOPING(lc); lc++) {
+-
+-		/* reset tst_count in case we are looping */
+-		tst_count = 0;
++  /*
++   * perform global setup for the test
++   */
++  setup();
++
++  for (lc = 0; TEST_LOOPING(lc); lc++) {
++
++    /* reset tst_count in case we are looping */
++    tst_count = 0;
++
++    /*
++     * When the process group having forked of a child
++     * and then it attached itself to another process
++     * group and tries to setsid
++     */
++    int s;
++
++    pthread_t master_tid;
++
++    s = pthread_create(&master_tid, NULL, do_master_thread, NULL);
++
++    if (s == 0)
++    {
++      printf("%s(): thread created fine ...\n", __FUNCTION__);
++      pthread_join(master_tid, NULL);
++    }
++    else
++    {
++      printf("%s(): thread created error ...\n", __FUNCTION__);
```

Comment:
please remove the print statements. use tst_resm with TINFO for information messages.